src/core/Form/Radiobutton/Radiobutton.baseStyles.tsx

```diff
@@ -0,0 +1,159 @@
+import { css } from 'styled-components';
+import { withSuomifiTheme, TokensAndTheme } from '../../theme';
+import { disabledCursor } from '../../../components/utils/css';
+import { element, font } from '../../theme/reset';
+import { focus } from '../../theme/utils/focus';
+
+/* stylelint-disable no-descending-specificity */
+export const baseStyles = withSuomifiTheme(
+  ({ theme }: TokensAndTheme) => css`
+    ${element({ theme })}
+    ${font({ theme })('bodyText')}
+
+  &.fi-radiobutton {
+    & .fi-radiobutton_hintText {
+        display: block;
+        padding-left: 26px;
+        color: ${theme.colors.depthDark1};
+        ${theme.typography.bodyTextSmall};
+    }
+
+    & .fi-radiobutton_input {
+      opacity: 0;
+      position: absolute;
+      margin: 8px 0 0 3px;
+
+      + label {
+        position: relative;
+        display: inline-block;
+        cursor: pointer;
+        min-height: 27px;
+        line-height: 1.5em;
+        padding-left: 26px;
+
+        &::before {
+          content: '';
+          position: absolute;
+          display: inline-block;
+          left: 0;
+          top: 5px;
+          width: 16px;
+          height: 16px;
+          border-radius: 50%;
+          border: 1px solid ${theme.colors.depthBase};
+          background: transparent;
+        }
+
+        &::after {
+          content: '';
+          position: absolute;
+          display: inline-block;
+          left: 5px;
+          top: 10px;
+          border-radius: 50%;
+          width: 8px;
+          height: 8px;
+        }
+      }
+
+      &:checked {
+        + label:before {
+          border: 1px solid ${theme.colors.highlightBase};
+        }
+
+        + label:after {
+          background: ${theme.colors.highlightBase};
+        }
+      }
+
+      &:focus {
+        + label::before {
+          ${focus({ theme, noPseudo: true, variant: 'boxShadow' })}
+          border-radius: 50%;
+        }
+      }
+
+      &:disabled {
+        + label {
+          ${disabledCursor}
```

Comment:
Are cursors automatically handled for other elements / no need to add this elsewhere when disabled?

Reply:
I think it is sufficient and working as of tested with Brave-browser.

`label` basically is the only visible element there as the` input` element is hidden.
That background and "selected" _circles_  are done with `::after` and `::before` pseudos of the `label`.